android/src/com/mapswithme/maps/MwmApplication.java

```diff
@@ -161,7 +160,30 @@ public String getApkPath()
     }
   }
 
-  public String getDataStoragePath()
+  private static boolean isMapStorage(String path)
+  {
+    return !TextUtils.isEmpty(path) && Utils.directoryExists(path) &&
+           new File(path, "World.mwm").exists();
+  }
+
+  private String findMapStoragePath()
+  {
+    String res = mPrefs.getString(Constants.PREF_STORAGE_PATH, "");
```

Comment:
No, this value is stored in internal preferences also to be used as a primary pointer to current map directory.